### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by sinergise-anze**
This PR configures `jest`:

- `expect` is extended with methods `toHaveOrigin`, `toHaveQueryParamsValues` and `toHaveQueryParams`
- tests are run automatically through GitHub Actions
- a small test is written for `WmsLayer` (on a non-SentinelHub host, to avoid SH-specific parameters)
- along the way, two small bugs are fixed in `WmsLayer`:
  - parameters `transparent` and `showlogo` are removed
  - trailing ampersand (`&`) is removed

Note that mocking requests is not part of this PR - the test is done on `getMapUrl()` method which simply returns a string.

### CODE REVIEW

File: jest-setup.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    return {
+      message: () => `URL hostname should not have value [${expectedOrigin}], but it does`, // if .not is used
+      pass: true,
+    };
+  },
+});
+
+/* ************************ */
+
+function breakUrl(urlWithQueryParams: string) {
```
Comment (dgostencnik) [inline jest-setup.ts]: Perhapse parseUrl would be better name for this function.

File: package.json [hunk 2]
```diff
@@ -31,13 +34,24 @@
...
     "build-storybook": "build-storybook"
   },
+  "jest": {
+    "moduleFileExtensions": [
+      "ts",
+      "tsx",
+      "js"
+    ],
+    "transform": {
+      "\\.(ts|tsx)$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
```
Comment (dgostencnik) [inline package.json]: jest doesn't like preprocessor (anymore)
```
(WARN) Replace any occurrences of "ts-jest/dist/preprocessor.js" or  "<rootDir>/node_modules/ts-jest/preprocessor.js" 
in the 'transform' section of your Jest config with just "ts-jest".
```

File: jest-setup.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      if (params[k] === undefined) {
+        return {
+          message: () => `URL query parameter [${k}] should exist, but it doesn't`,
+          pass: false,
+        };
+      }
+    }
+    return {
+      message: () =>
+        `URL [${received}] should not include all of the parameters ${JSON.stringify(expectedParamsKeys)}, but it does`,
```
Comment (dgostencnik) [inline jest-setup.ts]: This is a bit confusing, but if I understand correctly this is used when with .not ?

File: src/layer/__tests__/WmsLayer.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    bbox: bbox,
+    fromTime: new Date(Date.UTC(2020, 1 - 1, 10, 0, 0, 0)), // 2020-01-10/2020-01-10
+    toTime: new Date(Date.UTC(2020, 1 - 1, 10, 23, 59, 59)),
+    width: 512,
+    height: 512,
+    format: MimeTypes.JPEG,
+  };
+  const imageUrl = layer.getMapUrl(getMapParams, ApiType.WMS);
+
+  expect(imageUrl).toHaveOrigin('https://proba-v-mep.esa.int');
```
Comment (dgostencnik) [inline src/layer/__tests__/WmsLayer.ts]: Should baseUrl also be tested somewhere?

File: jest-setup.ts (hunk 3 above)
Comment (sinergise-anze) [reply jest-setup.ts]: Yes, I agree it is a bit confusing - and your understanding is correct (or at least the same as mine ;) ). However this is not something I can do much about...

File: package.json (hunk 2 above)
Comment (sinergise-anze) [reply package.json]: Thank you, missed that! I have now also added `esModuleInterop` config option to suppress another warning, and fixed Axios interceptors (they expected `caches` to be defined which is not the case when running tests).

File: jest-setup.ts (hunk 1 above)
Comment (sinergise-anze) [reply jest-setup.ts]: Yes, much better, thank you.

File: src/layer/__tests__/WmsLayer.ts (hunk 4 above)
Comment (sinergise-anze) [reply src/layer/__tests__/WmsLayer.ts]: Maybe, though this PR and #22 are mostly about setting up the tools that we will need for testing, not so much about the tests themselves. I did add a test just to show that these tools work, but I would prefer to keep them small and we can complete them in the future, if you agree?

Comment (dgostencnik) [reply src/layer/__tests__/WmsLayer.ts]: OK

**Pull request merged**